### PR TITLE
added Cognito resources to cdk, updated README

### PR DESCRIPTION
**Pull request by gsechkin-aws**
A few notes: it looks like Johnny's earlier commits from other branches are included in this PR.

TODOs:
- Figure out why specifically UserPool does not get deleted during cdk destroy but everything else does
- Was not able to automate the following command: aws cognito-idp admin-set-user-password --user-pool-id "[YOUR_USER_POOL_ID]" --username "[USERNAME]" --password "[PASSWORD]" --permanent, will revisit when have some more bandwidth
- Automate the filling in fields for config files in CookieFactoryDemo folder
- README's still may need adjustments

### CODE REVIEW

File: src/workspaces/cookiefactoryv2/README.md [hunk 1]
```diff
@@ -56,21 +104,59 @@ This project walks you through the process of setting up the Bakersville Cookie
...
+
+## Configure Web Application
+
+First change directory to CookieFactoryDemo
+``` 
+cd ../CookieFactoryDemo 
+```
+
+1. Change `WORKSPACE_ID` in `src/config/sites.ts` to your workspace ID.
+2. Use the following CLI command to administratively set the password for your demo Cognito user, Fran. Your User_Pool_id can be found after your CDK deploy completed in the CloudFormationOutput (printed to console). The demo Username is "fran@cookiefactory". Be sure to create a password that meets the default Cognito password requirements (Lowercase letter, Uppercase letter, Number, Symbol, Length >= 8)
```
Comment (johnnyw-aws) [inline src/workspaces/cookiefactoryv2/README.md:117]: > User_Pool_id

consistency: looks like it's "UserPoolId" in the cdk output

File: src/workspaces/cookiefactoryv2/README.md [hunk 2]
```diff
@@ -56,21 +104,59 @@ This project walks you through the process of setting up the Bakersville Cookie
...
+```
+aws cognito-idp admin-set-user-password --user-pool-id "[YOUR_USER_POOL_ID]" --username "[USERNAME]" --password "[PASSWORD]" --permanent
```
Comment (johnnyw-aws) [inline src/workspaces/cookiefactoryv2/README.md:119]: maybe we can add `jq` as a dependency in the pre-reqs and then save the user-pool-id automatically for the user similar to in https://github.com/aws-samples/aws-iot-twinmaker-samples/tree/main/src/modules/insights

File: src/workspaces/cookiefactoryv2/README.md [hunk 3]
```diff
@@ -56,21 +104,59 @@ This project walks you through the process of setting up the Bakersville Cookie
...
+
+
+## Configure Web Application
+
+First change directory to CookieFactoryDemo
+``` 
+cd ../CookieFactoryDemo 
+```
+
+1. Change `WORKSPACE_ID` in `src/config/sites.ts` to your workspace ID.
```
Comment (johnnyw-aws) [inline src/workspaces/cookiefactoryv2/README.md:116]: fyi that sites.ts was refactored into a template file as well, we probably should merge the latest README changes in as you mentioned in a TODO (can be separate PR if easier but if so then this one should probably go into a separate branch than main)

File: src/workspaces/cookiefactoryv2/README.md (hunk 2 above)
Comment (johnnyw-aws) [inline src/workspaces/cookiefactoryv2/README.md:119]: > The demo Username is "fran@cookiefactory"
> --username "[USERNAME]"

shall we just pre-populate the value for the user?

File: src/workspaces/cookiefactoryv2/README.md [hunk 4]
```diff
@@ -56,21 +104,59 @@ This project walks you through the process of setting up the Bakersville Cookie
...
-          --context stackName="__FILL_IN__" \
-          --context iottwinmakerWorkspaceId="__FILL_IN__" \
-          --context iottwinmakerWorkspaceBucket="__FILL_IN__"
+          --context stackName="CookieFactoryDemo" \
+          --context iottwinmakerWorkspaceId="$WORKSPACE_ID" \
+          --context iottwinmakerWorkspaceBucket="$WORKSPACE_BUCKET_NAME"
         ```
-2. Delete any Cognito-related resources setup for the demo if needed
-
+2. Navigate to the AWS Cognito console and delete CookiefactoryUserPool in the "User pools" tab by selecting the name, clicking "delete", and confirming the name in the prompt.
```
Comment (johnnyw-aws) [inline src/workspaces/cookiefactoryv2/README.md:159]: can we provide the CLI command(s) in interim instead of having them go to console?

File: src/workspaces/cookiefactoryv2/README.md [hunk 5]
```diff
@@ -56,21 +104,59 @@ This project walks you through the process of setting up the Bakersville Cookie
...
+* Navigate to the application at `localhost:5000` (default) - Note: it may take a minute to load the first time
+
+### Build
+
+```shell
+npm run build
+```
+---
 
 ## Cleanup
```
Comment (johnnyw-aws) [inline src/workspaces/cookiefactoryv2/README.md:149]: fyi: I forgot to include workspace deletion steps in the original README, was thinking to provide console steps and tmdt steps (the `destroy` command worked well for me btw!) -- feel free to include if you have bandwidth (no worries if not)

File: src/workspaces/cookiefactoryv2/README.md (hunk 1 above)
Comment (gsechkin-aws) [reply src/workspaces/cookiefactoryv2/README.md:117]: will update output

File: src/workspaces/cookiefactoryv2/README.md (hunk 2 above)
Comment (gsechkin-aws) [reply src/workspaces/cookiefactoryv2/README.md:119]: I considered this but left it like this to show user that their username can be different, but I guess it makes most sense to pre-populate, will relook into automating the password again when I revisit this

File: src/workspaces/cookiefactoryv2/README.md (hunk 4 above)
Comment (gsechkin-aws) [reply src/workspaces/cookiefactoryv2/README.md:159]: Yes I'll change to cli for now. Ideally the CDK should destroy it, I still am unsure why only the user pool doesn't get deleted by CDK

File: src/workspaces/cookiefactoryv2/README.md (hunk 5 above)
Comment (gsechkin-aws) [reply src/workspaces/cookiefactoryv2/README.md:149]: "console and tmdt", as in both options?

File: src/workspaces/cookiefactoryv2/README.md (hunk 2 above)
Comment (gsechkin-aws) [reply src/workspaces/cookiefactoryv2/README.md:119]: I'll look into jq if I have some extra time, although I didn't see how it's used in that insights folder, is it in code or CLI commands?

Comment (johnnyw-aws) [reply src/workspaces/cookiefactoryv2/README.md:119]: thanks, it's for the CLI, e.g. under "Set environment variables for convenience." in the README in insights folder

it's basically just a tool to let you extract/manipulate JSON, like setting an environment variable equal to a specific part of the output (e.g. the user pool id) which can be retrieved using AWS CLI to fetch the CFN output values

File: src/workspaces/cookiefactoryv2/README.md (hunk 4 above)
Comment (johnnyw-aws) [reply src/workspaces/cookiefactoryv2/README.md:159]: wonder if could be related to termination protection: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cognito-userpool.html#cfn-cognito-userpool-deletionprotection

File: src/workspaces/cookiefactoryv2/README.md (hunk 5 above)
Comment (johnnyw-aws) [reply src/workspaces/cookiefactoryv2/README.md:149]: right, I just meant as 2 different ways to delete the workspace-related resources

creation in console is fairly straightforward IMO but deletion has some extra steps so tmdt may be easier, but since we have console to create we should probably be consistent and at least include console to delete too

File: src/workspaces/cookiefactoryv2/README.md (hunk 4 above)
Comment (gsechkin-aws) [reply src/workspaces/cookiefactoryv2/README.md:159]: That was my first guess too since deletion protection is enabled in our original instructions, but I explicitly disabled it in the cdk code here
line 403: `deletionProtection: false,`

File: src/workspaces/cookiefactoryv2/README.md (hunk 5 above)
Comment (gsechkin-aws) [reply src/workspaces/cookiefactoryv2/README.md:149]: Makes sense. Should we include tmdt workspace create too? This would require either passing in a blank TMDT file as --dir or add a partial version of that code directly in this folder.

Comment (johnnyw-aws) [reply src/workspaces/cookiefactoryv2/README.md:149]: sounds good, let's give it a try and have the console steps as an alternative, we can revise based on customer feedback
